rest/src/main/java/com/ericsson/bss/cassandra/ecchronos/rest/RepairManagementRESTImpl.java

```diff
@@ -64,32 +72,153 @@ public RepairManagementRESTImpl(RepairScheduler repairScheduler, OnDemandRepairS
     }
 
     @Override
+    @GetMapping(ENDPOINT_PREFIX_V2 + "/schedules")
+    public ResponseEntity<List<Schedule>> getSchedules(@RequestParam(required = false) String keyspace,
+                                                       @RequestParam(required = false) String table)
+    {
+        if (keyspace != null)
+        {
+            if (table != null)
+            {
+                List<Schedule> repairJobs = getScheduledRepairJobs(forTable(keyspace, table));
+                return ResponseEntity.ok(repairJobs);
+            }
+            else
+            {
+                List<Schedule> repairJobs = getScheduledRepairJobs(
+                        job -> keyspace.equals(job.getTableReference().getKeyspace()));
+                return ResponseEntity.ok(repairJobs);
+            }
+        }
+        else if (table == null)
+        {
+            List<Schedule> repairJobs = getScheduledRepairJobs(job -> true);
+            return ResponseEntity.ok(repairJobs);
+        }
+        throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+    }
+
+    @Override
+    @GetMapping(ENDPOINT_PREFIX_V2 + "/schedules/{id}")
+    public ResponseEntity<Schedule> getSchedules(@PathVariable String id, @RequestParam(required = false) boolean full)
+    {
+
+        UUID uuid;
+        try
+        {
+            uuid = UUID.fromString(id);
+        }
+        catch (IllegalArgumentException e)
+        {
+            return ResponseEntity.badRequest().build();
```

Comment:
Throw for all errors.
throw new ResponseStatusException(HttpStatus.BAD_REQUEST)

Reply:
All these throw comments will be fix with #301. Let us contain all those fixes to this patch instead of spreading it to this patch.